### ISSUE TITLE
parallel unroll mode compounds prediction errors unnecessarily in multi-step loop

### ISSUE THREAD

**Issue by montoneguglielmo** [facebookresearch/eb_jepa#10]
In the unroll method of jepa.py, the parallel unroll mode runs the predictor nsteps times, each time re-feeding its own outputs. This is intended to train the predictor to handle its own errors (a form of scheduled sampling). However, the way ground truth states are re-injected creates an unnecessary error cascade.

The problem:
After each iteration, only the first context_length real states are pinned — everything else is replaced with predictions:

predicted_states = torch.cat(
    (state[:, :, :context_length], predicted_states), dim=2
)

With context_length=2 and StateOnlyPredictor (which predicts from pairs of consecutive states), iteration 2 receives:

[s1, s2, s*(3), s*(4), s*(5), ...]

The predictor then forms pairs:
(s1, s2) → s**(3) — both real ✓
(s2, s*(3)) → s**(4) — one real, one predicted ✓
(s*(3), s*(4)) → s**(5) — both predicted ✗

But s3 is real ground truth, available at training time. The pair for predicting s**(5) should be (s3, s*(4)), not (s*(3), s*(4)). The current code uses a predicted state where a real one could and should be used.

Expected behavior:
In iteration 2, the predictor should use real states as the "anchor" of each pair whenever the real state is available, and only propagate a predicted state as the "right edge" — i.e., the prediction being carried forward. This would mean:

(s2, s*(3)) → s**(4) ✓
(s3, s*(4)) → s**(5) ✓ — s3 real, not s*(3) predicted
(s4, s*(5)) → s**(6) ✓

Impact:
The error compounds faster than necessary across positions in the sequence. Distant positions are predicted from two predicted states when only one predicted state should be involved. This likely degrades the quality of the multi-step training signal.

Suggested fix:
Rather than replacing all positions after context_length with predictions, the loop should interleave real states at even positions and predicted states at odd positions (or more precisely: use the real state for the "left" element of each pair, and the predicted state only for the "right" element that is being propagated forward).
